l5kit/l5kit/environment/cle_metricset.py

```diff
@@ -0,0 +1,117 @@
+from abc import ABC, abstractmethod
+from typing import Any, DefaultDict, Dict, List
+
+from l5kit.cle import composite_metrics, metrics, validators
+from l5kit.cle.closed_loop_evaluator import ClosedLoopEvaluator, EvaluationPlan
+from l5kit.simulation.dataset import SimulationDataset
+from l5kit.simulation.unroll import SimulationOutputCLE, UnrollInputOutput
+
+
+class SimulationOutputGym(SimulationOutputCLE):
+    def __init__(self, scene_id: int, sim_dataset: SimulationDataset,
+                 ego_ins_outs: DefaultDict[int, List[UnrollInputOutput]],
+                 agents_ins_outs: DefaultDict[int, List[List[UnrollInputOutput]]]):
+        """This object holds information about the result of the simulation loop
+        for a given scene dataset in Gym
+
+        :param scene_id: the scene indices
+        :param sim_dataset: the simulation dataset
+        :param ego_ins_outs: all inputs and outputs for ego (each frame of each scene has only one)
+        :param agents_ins_outs: all inputs and outputs for agents (multiple per frame in a scene)
+        """
+        super(SimulationOutputGym, self).__init__(scene_id, sim_dataset, ego_ins_outs, agents_ins_outs)
+
+        # Required for Bokeh Visualizer
+        simulated_dataset = sim_dataset.scene_dataset_batch[scene_id]
+        self.tls_frames = simulated_dataset.dataset.tl_faces
+        self.agents_th = simulated_dataset.cfg["raster_params"]["filter_agents_threshold"]
+
+
+class CLEMetricSet(ABC):
```

Comment:
Rename this class to `CLEBaseMetricSet` as this is the base (abstract base class) class.